### PR TITLE
Updates for the models branch

### PR DESCRIPTION
**Pull request by lima-84**
Contains a series of changes in the models branch, organized separately in commits.

### CODE REVIEW

**Review by edumapurunga — APPROVED** [review]

File: pysid/identification/pemethod.py [hunk 1]
```diff
@@ -140,8 +140,8 @@ def arx(na, nb, nk, u, y, opt=0):
             ka += na[i, j]
     # Model
     m = polymodel('arx', A, B, None, None, None, nk, (u, y), nu, ny, 1)
-    m.setcov(V**2, V**2/Ny*dot(R, R.T), V**2/Ny)
```
Comment (edumapurunga) [inline pysid/identification/pemethod.py:143]: You need first to compute the prediction error. Add the following before this line
1. Compute e(t) = A(q) y(t) - B(q) (t). 
2. Estimate the noise covariance sig = 1/Ny e.T @ e (Verify if the dividing by Ny is necessary)
3. m.setcov(V**2/Ny, inv(R.T @ sig @ R), sig) 

Why are you adding R to setcov?

**Review by edumapurunga — CHANGES_REQUESTED** [review]

**Review by edumapurunga — APPROVED** [review]

File: pysid/identification/models.py [hunk 2]
```diff
@@ -50,3 +51,134 @@ def setcov(self, V, accuracy, ncov):
...
+        index = index + 1
+
+    model_str = model_str + "\nMODEL PROPERTIES:"
+    model_str = model_str + "\nTime delay: " + str(m.d)
+    model_str = model_str + "\nSample time: " + str(m.ts)
+
+    return model_str
+
+#%%
+# SISO Test
```
Comment (edumapurunga) [inline pysid/identification/models.py]: Remove this example from this commit.

**Review by edumapurunga — CHANGES_REQUESTED** [review]

File: pysid/identification/pemethod.py (hunk 1 above)
Comment (lima-84) [reply pysid/identification/pemethod.py:143]: I added a filtmat() function which will (hopefully) allow us to compute e(t) for MIMO ARX and ARMAX models. Needs more testing!

Due to sig = (e.T @ e) being an array, the scalar case (in which sig.shape = (1,1)) could not be handled by neither matmul() nor dot(), multi_dot(). To be clear: both functions didn't handle a (1,1) matrix as a scalar, which led me to manually check the dimension of sig and use matmul (@ operator).

The ARX sig computation should already be functional for MIMO systems. In regards to ARMAX, we must check the C matrix, considering that filtmat() expects C to be diagonal, which isn't the case yet.

File: pysid/identification/models.py (hunk 2 above)
Comment (lima-84) [reply pysid/identification/models.py]: Removed the example with rebase.

**Pull request merged**